src/services/fhirService.ts

```diff
@@ -588,24 +606,58 @@ const fhirService = {
     return recordService.deleteRecord(ownerId, resourceId);
   },
 
-  countResources(ownerId: string, params: Params = {}): Promise<number> {
-    const parameters = prepareSearchParameters({
-      ...params,
-      exclude_flags: [taggingUtils.generateAppDataFlagTag()],
+  getSearchCallWithFallbackIfNeeded(
+    ownerId: string,
+    countOnly: boolean,
+    params: Params = {}
+  ): Promise<any> {
+    const basicParameters = prepareSearchParameters({
+      params: {
+        ...params,
+        exclude_flags: [taggingUtils.generateAppDataFlagTag()],
+      },
+      useFallbackParams: false,
     });
-    return recordService.searchRecords(ownerId, parameters, true).then(result => result.totalCount);
+
+    if (params?.fhirVersion) {
+      const fallbackParameters = prepareSearchParameters({
+        params: {
+          ...params,
+          exclude_flags: [taggingUtils.generateAppDataFlagTag()],
+        },
+        useFallbackParams: true,
+      });
+      return Promise.all([
+        recordService.searchRecords(ownerId, basicParameters, countOnly),
+        recordService.searchRecords(ownerId, fallbackParameters, countOnly),
+      ]);
+    }
+    return recordService.searchRecords(ownerId, basicParameters, countOnly);
+  },
+
+  countResources(ownerId: string, params: Params = {}): Promise<number> {
+    return this.getSearchCallWithFallbackIfNeeded(ownerId, true, params).then(result =>
+      result.totalCount
+        ? result.totalCount
+        : parseInt(result[0].totalCount, 10) + parseInt(result[1].totalCount, 10)
```

Comment:
How about putting this part in the `getSearchCallWithFallbackIfNeeded` function so that it alwaws returns an object in the same format?

Reply:
See previous comment.